deepsensor/data/loader.py

```diff
@@ -1413,30 +1460,87 @@ def sample_sliding_window(
+            for y in np.arange(x1_max, x1_min, -dy):
+                for x in np.arange(x2_min, x2_max, dx):
+                    if y - x1_extend < x1_min:
+                        y0 = x1_min + x1_extend
+                    else:
+                        y0 = y
+                    if x + x2_extend > x2_max:
+                        x0 = x2_max - x2_extend
+                    else:
+                        x0 = x
+
+                    # bbox of x1_min, x1_max, x2_min, x2_max per patch
+                    bbox = [y0 - x1_extend, y0, x0, x0 + x2_extend]
+                    patch_list.append(bbox)
+
+        elif self.coord_directions['x1'] == False and self.coord_directions['x2'] == False:
+            for y in np.arange(x1_max, x1_min, -dy):
+                for x in np.arange(x2_max, x2_min, -dx):
+                    if y - x1_extend < x1_min:
+                        y0 = x1_min + x1_extend
+                    else:
+                        y0 = y
+                    if x - x2_extend < x2_min:
+                        x0 = x2_min + x2_extend
+                    else:
+                        x0 = x
+
+                    # bbox of x1_min, x1_max, x2_min, x2_max per patch
+                    bbox = [y0 - x1_extend, y0, x0 - x2_extend, x0]
+                    patch_list.append(bbox)
+
+        elif self.coord_directions['x1'] == True and self.coord_directions['x2'] == False:
+            for y in np.arange(x1_min, x1_max, dy):
+                for x in np.arange(x2_max, x2_min, -dx):
+                    if y + x1_extend > x1_max:
+                        y0 = x1_max - x1_extend
+                    else:
+                        y0 = y
+                    if x - x2_extend < x2_min:
+                        x0 = x2_min + x2_extend
+                    else:
+                        x0 = x
+
+                    # bbox of x1_min, x1_max, x2_min, x2_max per patch
+                    bbox = [y0, y0 + x1_extend, x0 - x2_extend, x0]
+                    patch_list.append(bbox)
+        else:
+            for y in np.arange(x1_min, x1_max, dy):
+                for x in np.arange(x2_min, x2_max, dx):
+                    if y + x1_extend > x1_max:
+                        y0 = x1_max - x1_extend
+                    else:
+                        y0 = y
+                    if x + x2_extend > x2_max:
+                        x0 = x2_max - x2_extend
+                    else:
+                        x0 = x
+
+                    # bbox of x1_min, x1_max, x2_min, x2_max per patch
+                    bbox = [y0, y0 + x1_extend, x0, x0 + x2_extend]
 
-                # bbox of x1_min, x1_max, x2_min, x2_max per patch
-                bbox = [y0, y0 + x1_extend, x0, x0 + x2_extend]
+                    patch_list.append(bbox)
 
-                patch_list.append(bbox)
+        # Remove duplicate patches while preserving order
+        seen = set()
+        unique_patch_list = []
```

Comment:
Possibly could leverage the characteristics of `set`s here to create a unique set of patches, but not sure if order is preserved in sets.